providers/s3/s3.go

```diff
@@ -310,12 +314,23 @@ func NewBucketWithConfig(logger log.Logger, config Config, component string) (*B
 		return nil, errors.Errorf("Initialize s3 client list objects version: Unsupported version %q was provided. Supported values are v1, v2", config.ListObjectsVersion)
 	}
 
+	var storageClass string
+	for _, k := range []string { amzStorageClass, strings.ToLower(amzStorageClass) } {
```

Comment:
I am not sure what we are trying to accomplish here 🤔 Do we need this loop?

Reply:
Ok, I had to read through the whole issue to understand this. Can we:

1. Comment why we have to do it.
2. Can we ensure we are prone to capitalizations? e.g. if user would put `X-Amz-Storage-class` we would miss it. Not a big deal but might be confusing for ppl.

For example instead we could iterate over `PutUserMetadata` and `string.ToLower` for both `amsStorageClass` and map key to check if we are matching it.

3. I wonder if we should not just add `StorageClass` option and fail if user pass it through `PutUserMetadata`. I guess less option is better,so maybe it's fine like this.

Reply:
I implemented item 2, and updated `TestParseConfig_CustomStorageClass` to check different cases of `X-Amz-Storage-Class` are all detected as the same.  

I didn't add a comment for item 1, because I though the new code is more obvious, but I can add another one of you think its needed.   

For item 3, I understood we agreed that not adding a new option was ok, but I'm open to adding a new one if you think that's best. An advantage of reusing `put_user_metadata` is that it doesn't require coming up with a new configuration setting that would make sense for other object stores

Reply:
Thanks for the clarification @juanrh, I think it is fine then, more of a nit from my side. It was not clear to me e.g. why we need to delete that metadata entry but understood it once I went through the issue.